### PR TITLE
[db] fix the query for non-fga migrated users

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description

Fixes the query for listing users that should be migrated by the FGA job.
At the moment the query is wrong and returns empty lists. As a consequence many users that haven't been active recently are not considered by FGA and can therefore not be found using our admin dashboard.

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at d43464b</samp>

Improve FGA migration query and add test case. The changes exclude built-in users from the query that selects users who need to migrate to a newer fine-grained access (FGA) version. The changes also add a unit test to verify the query logic and results.

</details>

## Related Issue(s)

Fixes #EXP-1021

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/gitpod-db/src/typeorm/user-db-impl.ts [hunk 1]
```diff
@@ -662,16 +663,13 @@ export class TypeORMUserDBImpl extends TransactionalDBImpl<UserDB> implements Us
 
     async findUserIdsNotYetMigratedToFgaVersion(fgaRelationshipsVersion: number, limit: number): Promise<string[]> {
         const userRepo = await this.getUserRepo();
-        const ids = (await userRepo
+        const users = await userRepo
             .createQueryBuilder("user")
-            .select(["id"])
-            .where({
-                fgaRelationshipsVersion: Not(Equal(fgaRelationshipsVersion)),
```
Comment (geropl) [inline components/gitpod-db/src/typeorm/user-db-impl.ts:669]: @svenefftinge Can you share some insight into what's not working here, e.g. how the resulting query looks like?

**Review by geropl — APPROVED** [review]
Code looks good, tests are green :heavy_check_mark:

File: components/gitpod-db/src/typeorm/user-db-impl.ts (hunk 1 above)
Comment (svenefftinge) [reply components/gitpod-db/src/typeorm/user-db-impl.ts:669]: no, I wrote a test and verified it failed and then used the pattern we use in other queries as well to make the test pass.

File: components/gitpod-db/src/typeorm/user-db-impl.ts [hunk 2]
```diff
@@ -662,16 +663,13 @@ export class TypeORMUserDBImpl extends TransactionalDBImpl<UserDB> implements Us
 
     async findUserIdsNotYetMigratedToFgaVersion(fgaRelationshipsVersion: number, limit: number): Promise<string[]> {
         const userRepo = await this.getUserRepo();
-        const ids = (await userRepo
+        const users = await userRepo
             .createQueryBuilder("user")
-            .select(["id"])
```
[the remark this answers is not in the document]
Comment (geropl) [reply components/gitpod-db/src/typeorm/user-db-impl.ts:667]: @svenefftinge I tried, and it's indeed this line that fails the method, `.select(["user.id"])` works. :roll_eyes:  So this is different to other querybuilder methods, which does not auto-namespace it's argument. And also, MySQL happily executes and retunrs and empty result... :see_no_evil:

**Pull request merged**